web/cypress/integration/utils/Common.ts

```diff
@@ -1,33 +1,69 @@
 import {camelCase} from 'lodash';
+import Bluebird = require('cypress/types/bluebird');
 import {Plugins} from '../../../src/constants/Plugins.constants';
 
 Cypress.on('uncaught:exception', err => !err.message.includes('ResizeObserver loop limit exceeded'));
 
 export const [{name, description}] = Plugins.REST.demoList;
 
-// eslint-disable-next-line import/no-mutable-exports
-export let testId = '';
+export const testRunPageRegex = /\/test\/(.*)\/run\/(.*)/;
 
-export const createTest = () => {
+function promisify(chain): Bluebird<unknown> {
+  return new Cypress.Promise((resolve, reject) => {
+    // We must subscribe to failures and bail. Without this, the Cypress runner would never stop
+    Cypress.on('fail', rejectPromise);
+    // // unsubscribe from test failure on both success and failure. This cleanup is essential
+    function resolvePromise(value) {
+      resolve(value);
+      Cypress.off('fail', rejectPromise);
+    }
+
+    function rejectPromise(error) {
+      reject(error);
+      Cypress.off('fail', rejectPromise);
+    }
+
+    chain.then(resolvePromise);
+  });
+}
+
+export function interceptTracePageApiCalls() {
+  cy.intercept({method: 'GET', url: '/api/tests/**/run/**'}).as('testRun');
+  cy.intercept({method: 'GET', url: '/api/tests/**'}).as('testObject');
+  cy.intercept({method: 'PUT', url: '/api/tests/**/run/**/dry-run'}).as('testRuns');
+}
+
+export function inteceptHomeApiCall() {
+  cy.intercept({method: 'GET', url: '/api/tests'}).as('testList');
+}
+
+export function waitForTracePageApiCalls() {
+  cy.wait('@testRun');
+  cy.wait('@testObject');
+  cy.wait('@testRuns', {timeout: 20000});
+}
```

Comment:
we are waiting for api call instead of waiting x milliseconds. this makes tests more reliable